src/ensemble_scheduler/ensemble_scheduler.cc

```diff
@@ -83,21 +83,23 @@ class RequestTracker {
     if (inflight_request_counter_ == 0) {
 #ifdef TRITON_ENABLE_STATS
       const auto& infer_stats = context_stats_aggregator_.ImmutableInferStats();
-      request_->ReportStatisticsWithDuration(
-          metric_reporter_, status_.IsOk(), compute_start_ns_,
-          infer_stats.compute_input_duration_ns_,
-          infer_stats.compute_infer_duration_ns_,
-          infer_stats.compute_output_duration_ns_);
-      if (status_.IsOk()) {
-        stats_aggregator_->UpdateInferBatchStatsWithDuration(
-            metric_reporter_, std::max(1U, request_->BatchSize()),
+      if (request_ != nullptr) {
```

Comment:
What was the reason for this change? Did you run into some newly introduced segfault case?

Reply:
Yes, I was facing a segfault here while testing the changes.
Used GDB to debug found calls were made when the request_ ptr was null, hence added the check

Reply:
Do you know what scenario or part of code newly introduced the segfault? It would be good to understand which code now makes requests possibly nullptr that wasn't happening before.

Reply:
Fixed :)
Detailed RCA
this happens only in case of dynamic batcher when using an ensemble model
FinishSkippedRequests gets called that in turn calls
RespondIfError which releases the request. (So far soo good)
But
In https://github.com/triton-inference-server/core/blob/627eb5ae9f65a9c146a99272a2ba91c01e0ac576/src/ensemble_scheduler/ensemble_scheduler.cc#L987
We set a callback
RequestComplete this in turn calls
DecrementCounter this in turn calls
ReportStatisticsWithDuration which in turn causes the crash.
The main "culprit" is calling RespondIfError with
true /* release_requests */

Reply:
Nice explanation, thanks for diving into it. So the `if (request_ == nullptr)` change is now essentially "unnecessary", right? But we can keep it in for safety purposes.

Reply:
Yes :)